interfaces/builtin/all_test.go

```diff
@@ -32,39 +32,39 @@ var _ = Suite(&AllSuite{})
 
 func (s *AllSuite) TestInterfaces(c *C) {
 	all := builtin.Interfaces()
-	c.Check(all, Contains, &builtin.BluezInterface{})
-	c.Check(all, Contains, &builtin.BoolFileInterface{})
-	c.Check(all, Contains, &builtin.BrowserSupportInterface{})
-	c.Check(all, Contains, &builtin.DbusInterface{})
-	c.Check(all, Contains, &builtin.DockerInterface{})
-	c.Check(all, Contains, &builtin.DockerSupportInterface{})
-	c.Check(all, Contains, &builtin.FramebufferInterface{})
-	c.Check(all, Contains, &builtin.FwupdInterface{})
-	c.Check(all, Contains, &builtin.GpioInterface{})
-	c.Check(all, Contains, &builtin.HidrawInterface{})
-	c.Check(all, Contains, &builtin.I2cInterface{})
-	c.Check(all, Contains, &builtin.IioInterface{})
-	c.Check(all, Contains, &builtin.IioPortsControlInterface{})
-	c.Check(all, Contains, &builtin.JoystickInterface{})
-	c.Check(all, Contains, &builtin.LocationControlInterface{})
-	c.Check(all, Contains, &builtin.LocationObserveInterface{})
-	c.Check(all, Contains, &builtin.LxdSupportInterface{})
-	c.Check(all, Contains, &builtin.MediaHubInterface{})
-	c.Check(all, Contains, &builtin.MaliitInterface{})
-	c.Check(all, Contains, &builtin.MirInterface{})
-	c.Check(all, Contains, &builtin.MprisInterface{})
-	c.Check(all, Contains, &builtin.PhysicalMemoryControlInterface{})
-	c.Check(all, Contains, &builtin.PhysicalMemoryObserveInterface{})
-	c.Check(all, Contains, &builtin.PulseAudioInterface{})
-	c.Check(all, Contains, &builtin.SerialPortInterface{})
-	c.Check(all, Contains, &builtin.ThumbnailerServiceInterface{})
-	c.Check(all, Contains, &builtin.TimeControlInterface{})
-	c.Check(all, Contains, &builtin.UDisks2Interface{})
-	c.Check(all, Contains, &builtin.UbuntuDownloadManagerInterface{})
-	c.Check(all, Contains, &builtin.Unity8Interface{})
-	c.Check(all, Contains, &builtin.UpowerObserveInterface{})
-	c.Check(all, Contains, &builtin.UhidInterface{})
-	c.Check(all, Contains, &builtin.Unity7Interface{})
+	c.Check(all, DeepContains, &builtin.BluezInterface{})
+	c.Check(all, DeepContains, &builtin.BoolFileInterface{})
```

Comment:
yes, it's 1 and something has changed, it's import to note that those structs are zero-sized.

The langref has this to say (at the very end):

"Two distinct zero-size variables may have the same address in memory."

may not must, so both <1.8 and 1.8 are doing acceptable things but the test was broken